src/scripts/main.js

```diff
@@ -1,11 +1,21 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
```

Comment:
Ensure that the elements with IDs 'login', 'password', and 'submit' exist in the HTML. If any of these elements do not exist, the `getElementById` calls will return `null`, leading to errors when trying to add event listeners.